### PR TITLE
feat(rank-collective): added new assoc. types

### PR DESCRIPTION
**Pull request by PraetorP**
This PR is a migration of the https://github.com/paritytech/substrate/pull/14577.

The PR added associated types (`AddOrigin` & `RemoveOrigin`) to `Config`. It allows you to decouple types and areas of responsibility, since at the moment the same types are responsible for adding and promoting(removing and demoting). This will improve the flexibility of the pallet configuration.

```
/// The origin required to add a member.
type AddOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = ()>;

/// The origin required to remove a member. The success value indicates the
/// maximum rank *from which* the removal may be.
type RemoveOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Rank>;
```
To achieve the backward compatibility, the users of the pallet can use the old type via the new morph:

```
type AddOrigin = MapSuccess<Self::PromoteOrigin, Ignore>;
type RemoveOrigin = Self::DemoteOrigin;
```

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3468835

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3469979

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3472411

File: cumulus/parachains/runtimes/collectives/collectives-polkadot/src/fellowship/mod.rs [hunk 1]
```diff
@@ -111,7 +111,12 @@ impl pallet_ranked_collective::Config<FellowshipCollectiveInstance> for Runtime
 
 	#[cfg(not(feature = "runtime-benchmarks"))]
 	// Promotions and the induction of new members are serviced by `FellowshipCore` pallet instance.
```
Comment (bkchr) [inline cumulus/parachains/runtimes/collectives/collectives-polkadot/src/fellowship/mod.rs:113]: The comment is out of place now.

File: substrate/primitives/runtime/src/traits.rs [hunk 2]
```diff
@@ -540,6 +540,9 @@ morph_types! {
 	/// Morpher to disregard the source value and replace with another.
 	pub type Replace<V: TypedGet> = |_| -> V::Type { V::get() };
 
+	/// Morpher to disregard the source value.
+	pub type Ignore = |_| -> () {};
+
```
Comment (bkchr) [inline substrate/primitives/runtime/src/traits.rs:545]: The naming is not really great and I don't think that we need this as a type here.

**Review by bkchr — APPROVED** [review]
Please merge master and provide some `prdoc`.

File: substrate/primitives/runtime/src/traits.rs (hunk 2 above)
Comment (ggwpez) [reply substrate/primitives/runtime/src/traits.rs:545]: Changed to `ReplaceWithDefault`.

**Comment by ggwpez** [comment]
Cannot push. Going to re-open.

**Pull request closed without merging**